### PR TITLE
UART Driver

### PR DESCRIPTION
**Pull request by Steven141**
### Summary

Implemented UART driver.

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: firmware/shared/src/hw/hw_uart.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+void hw_uart_receive_dma(UART *uart, uint8_t *pData, uint8_t size);
+
+/**
+ * Transmits an amount of data in polling mode (blocking).
+ * @param uart Pointer to a UART structure.
+ * @param pData Pointer to data buffer (u8 or u16 data elements).
+ * @param Size Amount of data elements (u8 or u16) to be transmitted.
+ * @param timeout Timeout duration
+ */
+void hw_uart_transmit_poll(UART *uart, uint8_t *pData, uint8_t size, uint32_t timeout);
```
Comment (Evanyl) [inline firmware/shared/src/hw/hw_uart.h:38]: nit: i don't like this naming scheme, can u rename all the variables in this file

Comment (gtaharaedmonds) [reply firmware/shared/src/hw/hw_uart.h:38]: yeah strong effort @Steven141 can you please close this PR and try again

Comment (jpatch22) [reply firmware/shared/src/hw/hw_uart.h:38]: I third this!

**Review by gtaharaedmonds — APPROVED** [review]
LGTM ⭐

**Pull request merged**